File: AssistantPlus/Application/View/CoreCommandSender/Configuration.cs
#pragma warning disable 0,

using Twinning.AssistantPlus;
using Twinning.AssistantPlus.Utility;

namespace Twinning.AssistantPlus.View.CoreCommandSender {

	public record PresetConfiguration {
		public String                     Name     { get; set; } = default!;
		public Dictionary<String, Object> Argument { get; set; } = default!;
	}

	public enum ArgumentType {
		Boolean,
		Integer,
		Floater,
		String,
		Size,
		Path,
	}

	public record ArgumentConfiguration {
		public String        Identifier { get; set; } = default!;
		public String        Name       { get; set; } = default!;
		public ArgumentType  Type       { get; set; } = default!;
		public List<Object>? Option     { get; set; } = default!;
	}

	public record MethodConfiguration {
		public String                      Identifier { get; set; } = default!;
		public String                      Name       { get; set; } = default!;
		public String                      Icon       { get; set; } = default!;
		public List<ArgumentConfiguration> Argument   { get; set; } = default!;
		public List<String>?               Batch      { get; set; } = default!;
		public List<PresetConfiguration?>  Preset     { get; set; } = default!;
	}

	public record MethodGroupConfiguration {
		public String                    Identifier { get; set; } = default!;
		public String                    Name       { get; set; } = default!;
		public String                    Icon       { get; set; } = default!;
		public List<MethodConfiguration> Item       { get; set; } = default!;
	}

	public record Configuration {
		public List<MethodGroupConfiguration> Method { get; set; } = default!;
	}

}

File: module/assistant/lib/view/core_command_sender/configuration.dart
import '/common.dart';

// ----------------

class PresetConfiguration {
  String              name;
  Map<String, Object> argument;
  PresetConfiguration({
    required this.name,
    required this.argument,
  });
}

enum ArgumentType {
  boolean,
  integer,
  floater,
  string,
  size,
  path,
}

class ArgumentConfiguration {
  String        identifier;
  String        name;
  ArgumentType  type;
  List<Object>? option;
  ArgumentConfiguration({
    required this.identifier,
    required this.name,
    required this.type,
    required this.option,
  });
}

class MethodConfiguration {
  String                      identifier;
  String                      name;
  String                      icon;
  List<ArgumentConfiguration> argument;
  List<String>?               batch;
  List<PresetConfiguration?>  preset;
  MethodConfiguration({
    required this.identifier,
    required this.name,
    required this.icon,
    required this.argument,
    required this.batch,
    required this.preset,
  });
}

class MethodGroupConfiguration {
  String                    identifier;
  String                    name;
  String                    icon;
  List<MethodConfiguration> item;
  MethodGroupConfiguration({
    required this.identifier,
    required this.name,
    required this.icon,
    required this.item,
  });
}

class Configuration {
  List<MethodGroupConfiguration> method;
  Configuration({
    required this.method,
  });
}
